libraries/botbuilder-core/botbuilder/core/adapters/test_adapter.py

```diff
@@ -151,10 +177,71 @@ def id_mapper(activity):
                 if len(arg) == 4:
                     timeout = arg[3]
             await self.test(arg[0], arg[1], description, timeout)
+    
+    def add_user_token(self, connection_name: str, channel_id: str, user_id: str, token: str, magic_code: str = None):
+        key = UserToken()
+        key.channel_id = channel_id
+        key.connection_name = connection_name
+        key.user_id = user_id
+        key.token = token
+
+        if not magic_code:
+            self._user_tokens.append(key)
+        else:
+            mc = TokenMagicCode()
+            mc.key = key
+            mc.magic_code = magic_code
+            self._magic_codes.append(mc)
+
+    async def get_user_token(self, context: TurnContext, connection_name: str, magic_code: str= None) -> TokenResponse:
+        key= UserToken()
+        key.channel_id = context.activity.channel_id
+        key.connection_name = connection_name
+        key.user_id = context.activity.from_property.id
+
+        if (magic_code):
+            magic_code_record = list(filter(lambda x: key.equals_key(x.key), self._magic_codes))
+            if (magic_code_record and len(magic_code_record) > 0 and magic_code_record[0].magic_code == magic_code):
+                # move the token to long term dictionary
```

Comment:
In general we prefer writing comments as english sentences, i.e. start sentence with uppercase: Move the token to long term dictionary.

Reply:
solved